includes/class-stats.php

```diff
@@ -0,0 +1,80 @@
+<?php
+/**
+ * File containing the class WP_Job_Manager_Stats
+ *
+ * @package wp-job-manager
+ */
+
+namespace WP_Job_Manager;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * This class is responsible for initializing all aspects of stats for wpjm.
+ */
+class Stats {
+	use Singleton;
+
+	/**
+	 * Constructor.
+	 */
+	private function __construct() {
+		$this->init();
+	}
+
+	/**
+	 * Do initialization of all the things needed for stats.
+	 */
+	public function init() {
+		$this->initialize_wpdb();
+	}
+
+	/**
+	 * Initialize the alias for the stats table on the wpdb object.
+	 *
+	 * @return void
+	 */
+	private function initialize_wpdb() {
+		global $wpdb;
+		if ( isset( $wpdb->job_manager_stats ) ) {
+			return;
+		}
+		$wpdb->job_manager_stats = $wpdb->prefix . 'job_manager_stats';
+		$wpdb->tables[]          = 'job_manager_stats';
+	}
+
+	/**
+	 * Migrate the stats table to the latest version.
+	 *
+	 * @return void
+	 */
+	public function migrate() {
+		global $wpdb;
+		$collate = $wpdb->get_charset_collate();
+		require_once ABSPATH . 'wp-admin/includes/upgrade.php';
+		\dbDelta(
+			[
+				"CREATE TABLE {$wpdb->job_manager_stats} (
```

Comment:
I know this isn't clear at first, but `dbDelta` already handles that for us, and I think it doesn't support having `IF NOT EXISTS` on the table name (it will consider the `IF NOT EXISTS`  part of the table name, which I don't think we want to have). See how it handles that here: https://github.com/WordPress/WordPress/blob/27348531f3fdf048809da977fced3f40b90259de/wp-admin/includes/upgrade.php#L2836-L2840

You can [see here](https://github.com/WordPress/WordPress/blob/27348531f3fdf048809da977fced3f40b90259de/wp-admin/includes/schema.php#L56-L237) that no WP tables use `IF NOT EXISTS` when creating the schema (those are passed to `dbDelta`, too). So I'm inclined to not add them. WDYT?